C/services/south/ingest.cpp

```diff
@@ -499,14 +503,35 @@ void Ingest::processQueue()
 				m_failCnt = 0;
 				std::map<std::string, int>		statsEntriesCurrQueue;
 				AssetTracker *tracker = AssetTracker::getAssetTracker();
+				StorageAssetTracker *satracker = StorageAssetTracker::getStorageAssetTracker();
+				if ( satracker == nullptr)
+                                {
+                                        Logger::getLogger()->error("%s:%d could not initialize satracker ", __FILE__, __LINE__);
+					return;
+                                }
 
 				string lastAsset = "";
 				int *lastStat = NULL;
+				std::map <std::string , std::set<std::string> > assetDatapointMap;
```

Comment:
How do we handle the case and store information in this map container where one reading for asset a1 has DPs - dp1, dp2 & dp3. And then another reading for same asset a1 has DPs - dp3, dp4 & dp5.